File: algorithms/Python/searching/binarySearch.py
def binarySearch(arr, l, r, x):
    """binarySearch Algorithm in python.

    :arr: the input array
    :x: value to search for
    :returns: the index of the value

    """
    while l <= r:
        mid = (l + r) // 2
        if arr[mid] == x:
            return mid
        elif arr[mid] > x:
            r = mid - 1
        else:
            l = mid + 1
    return -1

arr = [ 2, 3, 4, 10, 40 ]
x = 10

# Function call
result = binarySearch(arr, 0, len(arr)-1, x)

if result != -1:
    print ("Element is present at index % d" % result)
else:
    print ("Element is not present in array")

File: algorithms/javascript/searching/binarySearch.js
const binarySearch = (list, value) => {
    let l = 0;
    let r = list.length - 1;
    let mid;

    while (l <= r) {
        mid = Math.floor((l + r)/2);
        if (list[mid] == value) return mid;
        else if (list[mid] > value) r = mid - 1;
        else l = mid + 1;
    }
    return -1;
}

console.log(binarySearch([1, 2, 4, 5, 6, 33, 57], 8));
